### ISSUE TITLE
implied echo and $( )

### ISSUE THREAD

**Issue by mvolkmann** [nushell/nushell#3182]
**Is your feature request related to a problem? Please describe.**
Just an idea to make code more concise ...

**Describe the solution you'd like**
It would be cool if placing a value before a pipe could imply an echo of the value.
For example, instead of writing this:

```bash
    some-custom-command $(echo [$v1 $v2 $v3] | str collect)
```

we could write this:

```bash
    some-custom-command $([$v1 $v2 $v3] | str collect)
```

Even cooler would be if the `$( )` could be implied like this:

```bash
    some-custom-command [$v1 $v2 $v3] | str collect
```

But perhaps there are reasons I'm not considering why this is not possible.

**Comment by jonathandturner** [comment]
partially fixed by #3389

**Comment by jonathandturner** [comment]
I think the subexpression work we've been does the bulk of simplification here.

We wouldn't be able to do the third example as it would be visually ambiguous to new readers of code when the pipeline would bind to a command. 

Going ahead and closing.
